src/Signature/SignatureV4.php

```diff
@@ -146,6 +152,9 @@ protected function getPayload(RequestInterface $request)
 
     protected function getPresignedPayload(RequestInterface $request)
     {
+        if ($this->unsigned){
```

Comment:
This will only have an effect on presigned requests. I think you meant to put this block inside the `getPayload` method.